DevelopGitHub/GitHubExtension/GitHubExtention.Preferences.WebApi/Queries/PreferencesQuery.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GitHubExtension.Preferences.DAL.Model;
+
+namespace GitHubExtention.Preferences.WebApi.Queries
+{
+    class PreferencesQuery : IPreferencesQuery
+    {
+        private readonly PreferencesContext _context;
+
+        public PreferencesQuery(PreferencesContext context)
+        {
+            _context = context;
+        }
+
+        public IOrderedQueryable<User> Users
+        {
+            get { return _context.Users; }
+        }
+
+        public User GetUser(string id)
+        {
+            return _context.Users.FirstOrDefault(x => x.Id == id);
+        }
+
+        public void UpdateAvatarUrl(User user, string url)
+        {
+            user.AvatarUrl = url;
```

Comment:
May be it's better to pass User entity with updated state? Because now I don't see any reson to pass this additional parameter as URL.